### PR TITLE
Feat/Rosetta NEP-141

### PR DESCRIPTION
**Pull request by firatNEAR**
Rosetta NEP-141 implementation 
Includes: 
* NEP-141 tokens on `/block` and `/account/balance` endpoints based on provided NEP-141 token metadata in `config.json`.
* Rosetta pytest for sanity check including NEP-141 txs

### CODE REVIEW

File: chain/rosetta-rpc/src/adapters/nep141.rs
Comment (jakmeier) [inline chain/rosetta-rpc/src/adapters/nep141.rs:36]: (optional) nit: you should be able to avoid the `Vec::clone` and instead only clone the strings.
```suggestion
                    currencies.iter().cloned().collect::<HashMap<String, Currency>>();
```

File: chain/rosetta-rpc/src/adapters/nep141.rs
Comment (jakmeier) [inline chain/rosetta-rpc/src/adapters/nep141.rs:39]: (optional) nit: allocating a new string shouldn't be necessary
```suggestion
                        currency_map.get(outcome.outcome.executor_id.as_str())
```

File: chain/rosetta-rpc/src/models.rs
Comment (jakmeier) [inline chain/rosetta-rpc/src/models.rs:1378]: I believe an `i64` is too small? According to NEP-141, the amount is stored in `u128`, right? So delta could be up to  +/- `u128`, which means not even `i128` is large enough :(

File: chain/rosetta-rpc/src/adapters/nep141.rs
Comment (jakmeier) [inline chain/rosetta-rpc/src/adapters/nep141.rs:47]: For the parsing of up to 128-bit unsigned integer, this needs special handling. You can probably use the existing `impl DecType for u128` in `core/primitives-core/src/serialize.rs`.

But it won't work with negative values, you should use SignedDiff or something like that.

File: chain/rosetta-rpc/src/adapters/nep141.rs
Comment (jakmeier) [inline chain/rosetta-rpc/src/adapters/nep141.rs:64]: again, this won't parse all valid values

File: chain/rosetta-rpc/src/adapters/nep141.rs
Comment (jakmeier) [inline chain/rosetta-rpc/src/adapters/nep141.rs:98]: Nothing wrong with `from_str` but it's usually preferred to use `contract_address.parse()?` which is shorter since you don't need to type the type.

File: chain/rosetta-rpc/src/adapters/nep141.rs
Comment (jakmeier) [inline chain/rosetta-rpc/src/adapters/nep141.rs:100]: similarly here, nothing wrong with `from` but `args.into()` could infer the type for you

File: chain/rosetta-rpc/src/adapters/nep141.rs
Comment (jakmeier) [inline chain/rosetta-rpc/src/adapters/nep141.rs:123]: (optional)
You are dropping a very useful [serde_json Error](https://docs.rs/serde_json/latest/serde_json/struct.Error.html) here. Consider displaying it in the error message to help debugging any potential errors. For example when a custom FT contract prints invalid JSON events.

Ideal would be to add a parsing error in `ErrorKind` and add the serde errer as a `#[source]`.

But even just including it in the string of `InternalInvariantError` could be a benefit IMO.

But again - optional.

File: chain/rosetta-rpc/src/lib.rs
Comment (jakmeier) [inline chain/rosetta-rpc/src/lib.rs:402]: You could reduce indentation by 1 level if you use the `if let Some(currencies) = currencies { ... } else { ... }` syntax.

File: pytest/tests/sanity/rosetta.py
Comment (jakmeier) [inline pytest/tests/sanity/rosetta.py:713]:
```suggestion
        logger.info(f'Tx-result: {json_res}')
```

**Review by jakmeier — CHANGES_REQUESTED** [review]
I had a cursory look at the entire PR and left my thoughts.

The only thing that definitely needs to change before we can merge it is the upgrade schedule. Maybe the parsing of large values is also quite important. But otherwise it's mostly unimportant style suggestions.

But for the Rosetta side of things, making sure this is correct etc, I think you might want someone from your team to review it? Or are you just looking to get this synced to nearcore/master and it has already been reviewed and tested before? @firatNEAR

**Review by jakmeier — APPROVED** [review]
My concerns have been resolved. Your decision if someone else should review for the rosetta RPC logic

**Comment by jakmeier** [comment]
Well, my approval no longer holds the necessary power to merge... You need to find an active code owner to approve.

I don't think anyone in the protocol team is familiar with the Rosetta RPC though, so I don't know who to recommend. Maybe @akhi3030 can help to find someone or simply "reactive" my approval with their own stamp.

**Review by akhi3030 — APPROVED** [review]

**Comment by akhi3030** [comment]
Approved to unblock.

**Pull request merged**
